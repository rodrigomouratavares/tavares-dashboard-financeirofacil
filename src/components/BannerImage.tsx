import styled from 'styled-components'

export const BannerImage = styled.div`
  background-image: url(/login-image-tavares.svg);
  background-size: cover;
  height: 100vh;
  width: 50vw;
`
