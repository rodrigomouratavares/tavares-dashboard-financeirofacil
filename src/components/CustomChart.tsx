import { useMemo } from 'react'
import { useTheme } from 'styled-components'
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  BarElement,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js'
import { Line, Bar } from 'react-chartjs-2'
import { CustomChartProps } from '@/types'

ChartJS.register(
  CategoryScale,
  LinearScale,
  BarElement,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
)

function CustomChart({ data, labels, type }: CustomChartProps) {
  const theme = useTheme()

  const options = useMemo(
    () => ({
      responsive: true,
      scaleShowVerticalLines: false,
      scales: {
        x: {
          border: { display: false },
          grid: { display: false },
          ticks: { color: theme.typographies.subtitle },
        },
        y: {
          border: { display: false },
          grid: { color: theme.appDefaultStroke },
          ticks: { color: theme.typographies.subtitle },
        },
      },
      plugins: {
        legend: { display: false },
      },
    }),
    [theme]
  )

  const chartData = useMemo(
    () => ({
      labels,
      datasets: [
        {
          data,
          borderColor: 'rgb(12, 112, 242)',
          backgroundColor: 'rgba(12, 112, 242, 1)',
        },
      ],
    }),
    [labels, data]
  )

  return type === 'bar' ? (
    <Bar options={options} data={chartData} />
  ) : (
    <Line options={options} data={chartData} />
  )
}

export default CustomChart
