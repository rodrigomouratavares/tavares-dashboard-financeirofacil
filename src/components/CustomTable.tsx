import styled from 'styled-components'
import { CustomTableProps } from '@/types'
import { pxToRem } from '@/utils'

export const TablleWrapper = styled.header`
  overflow-x: auto;
  width: 100%;
  table {
    width: 100%;
    border-collapse: collapse;

    th,
    td {
      height: ${pxToRem(48)};
      padding: 0 ${pxToRem(8)} 0 0;
      text-align: left;
      &:last-child{
      text-align:right;
      padding:0;
      
      }
    }
      th {
        color:  background-color: ${(props) => props.theme.typographies.subtitle};
        font-weight:600;

      }

        tr{
        border-bottom : ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
         &:last-child{
      text-align:right;
         border-bottom:none;        }
  }
`
function CustomTable(props: CustomTableProps) {
  const { headers, rows } = props
  return (
    <TablleWrapper>
      <table>
        <thead>
          <tr>
            {headers.map((header, index) => (
              <th key={index}>{header}</th>
            ))}
          </tr>
        </thead>

        <tbody>
          {rows.map((row, rowIndex) => (
            <tr key={rowIndex}>
              {row.map((cell, cellIndex) => (
                <td key={cellIndex}>{cell}</td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </TablleWrapper>
  )
}

export default CustomTable
