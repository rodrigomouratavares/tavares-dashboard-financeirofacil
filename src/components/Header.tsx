import styled from 'styled-components'
import { Logo } from '@/components'
import { Link } from 'react-router-dom'
import { Avatar, Box, Container } from '@mui/material'
import { pxToRem } from '@/utils'

export const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`
function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={25} width={100} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="TAVARES AVATAR"
              src="/Avatar.svg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            ></Avatar>
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
