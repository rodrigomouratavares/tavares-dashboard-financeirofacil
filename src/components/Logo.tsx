import styled from 'styled-components'
import { pxToRem } from '@/utils'

export const Logo = styled.figure<{ height: number; width: number }>`
  background-image: url(/${(props) => props.theme.appLogo});
  background-size: cover;
  height: ${(props) => pxToRem(props.height)};
  width: ${(props) => pxToRem(props.width)};
`
