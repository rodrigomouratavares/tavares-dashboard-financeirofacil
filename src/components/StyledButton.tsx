import styled from 'styled-components'
import { ButtonProps } from '@/types'
import { pxToRem } from '@/utils'
export const StyledButton = styled.button<ButtonProps>`
  border-radius: ${pxToRem(8)};
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: ${pxToRem(16)};
  font-weigh: bold;
  height:${pxToRem(50)};
  padding: 0 ${pxToRem(16)};
  transition: background-color 0.3s;
  width: 100%;

  &.primary {
    background-color: ${(props) => props.theme.buttons.primary};
    color: ${(props) => props.theme.buttons.primaryColor};
    &:hover {
      background-color: ${(props) => props.theme.buttons.primaryHover};
    }
  }

  &.alert {
     background-color: ${(props) => props.theme.buttons.alert};
    color: ${(props) => props.theme.buttons.alertColor};
    &:hover {
      background-color: ${(props) => props.theme.buttons.alertHover};
  }

    &.bordeless-alert {
     background-color: none;
    color: ${(props) => props.theme.buttons.alert};
    height:0;
    padding: 0;
    &:hover {
      background-color: ${(props) => props.theme.buttons.alertHover};
  }

  &.disabled {
     background-color: ${(props) => props.theme.buttons.disabled};
    color: ${(props) => props.theme.buttons.disabledColor};
    cursor:not-allowed
    &:hover {
      background-color: ${(props) => props.theme.buttons.disabled};
      color: ${(props) => props.theme.buttons.disabledColor};
  }
`
