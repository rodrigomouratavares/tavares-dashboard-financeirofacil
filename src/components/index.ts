export { default as FormComponent } from './FormComponent' //procurar entender por que dessa forma
export { default as Header } from './Header'
export * from './BannerImage'
export * from './Logo'
export * from './StyledButton'
export * from './StyledInput'
export * from './Typographies'
