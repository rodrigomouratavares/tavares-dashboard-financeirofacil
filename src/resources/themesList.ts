import { darkTheme, lightTheme } from '@/styles'

export const themesList = [
  { name: 'Light Theme', theme: lightTheme },
  { name: 'Dark Theme', theme: darkTheme },
]
