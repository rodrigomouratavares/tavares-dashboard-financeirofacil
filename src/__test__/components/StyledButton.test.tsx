import { render } from '@testing-library/react'
import { StyledButton } from '@/components'
import { DefaultTheme, ThemeProvider } from 'styled-components'

import { themesList } from '@/resources/themesList'

describe('StyledButton Component', () => {
  const renderComponent = (
    theme: DefaultTheme,
    className: string,
    props = {}
  ) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledButton className={className} {...props} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot with the alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with the primary class', () => {
        const { asFragment } = renderComponent(theme, 'primary')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with the borderless-alert class', () => {
        const { asFragment } = renderComponent(theme, 'borderless-alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match the snapshot with the disabled status', () => {
        const { asFragment } = renderComponent(theme, 'primary', {
          disabled: true,
        })
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
