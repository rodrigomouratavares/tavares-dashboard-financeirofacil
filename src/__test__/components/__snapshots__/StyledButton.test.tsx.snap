// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component Dark Theme should match the snapshot with the alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component Dark Theme should match the snapshot with the borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component Dark Theme should match the snapshot with the disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Component Dark Theme should match the snapshot with the primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6D7B8E;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Component Light Theme should match the snapshot with the alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCCCCC;
  color: #666666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCCCCC;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component Light Theme should match the snapshot with the borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCCCCC;
  color: #666666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCCCCC;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component Light Theme should match the snapshot with the disabled status 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCCCCC;
  color: #666666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCCCCC;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Component Light Theme should match the snapshot with the primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 0.75rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0C70F2;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0061DE;
}

.c0.alert {
  background-color: #E80000;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #D80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #E80000;
  height: 0;
  padding: 0;
}

.c0.borderless-alert:hover {
  color: #D80000;
}

.c0:disabled {
  background-color: #CCCCCC;
  color: #666666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #CCCCCC;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;
